Makefile

```diff
@@ -246,7 +247,7 @@ develop:
 		-fsanitize=signed-integer-overflow -fsanitize=bounds \
 		-fsanitize=object-size -fsanitize=bool -fsanitize=enum \
 		-fsanitize=alignment -fsanitize=null -fsanitize=address" \
-		CFLAGS="-ggdb3 -Og -fno-omit-frame-pointer -fno-optimize-sibling-calls" \
+		CFLAGS="-ggdb3 -O0 -fno-omit-frame-pointer -fno-optimize-sibling-calls" \
```

Comment:
Why not `-Og` for C as well as C++? Was this just for testing?

> Optimize debugging experience. `-Og` should be the optimization level of choice for the standard edit-compile-debug cycle, offering a reasonable level of optimization while maintaining fast compilation and a good debugging experience. It is a better choice than `-O0` for producing debuggable code because some compiler passes that collect debug information are disabled at `-O0`.

Like `-O0`, `-Og` completely disables a number of optimization passes so that individual options controlling them have no effect. Otherwise `-Og` enables all `-O1` optimization flags except for those that may interfere with debugging.

-- <https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html>

Reply:
I had to disable all optimisations because I was getting this:

![Registers being optimised out in GDB](https://user-images.githubusercontent.com/15271137/193401382-e8c59f0e-97d2-4821-9e09-14b20d691b70.png)

...but it should not have been committed.